data/lib/api/match/match_model.dart

```diff
@@ -356,7 +356,7 @@ extension TeamStatExtension on List<MatchModel> {
       wickets += team.wicket;
 
       battingAverageTotal +=
-          opponentTeam.wicket > 0 ? team.run / opponentTeam.wicket : 0;
+          team.wicket > 0 ? team.run / team.wicket : team.run;
       if (team.wicket > 0) {
         bowlingAverageTotal += opponentTeam.run / team.wicket;
```

Comment:
@cp-mayank  
**team.run** -->  run made by team trough batting.
**team.wicket** --> wicket taken by team through bowling.

battingAVG  = runMadeByTeam / HowManyTimesGotDismissed

which is  team.run / **opponentTeam**.wicket.

let's check and revert changes accordingly.